pkg/contracts/src/CVStrategy.sol

```diff
@@ -963,4 +874,95 @@ contract CVStrategy is BaseStrategy, IPointStrategy, ERC165 {
         emit MinThresholdPointsUpdated(_minThresholdPoints, minThresholdPoints_);
         _minThresholdPoints = minThresholdPoints_;
     }
+
+    function disputeProposal(uint256 proposalId, string calldata context, bytes calldata _extraData)
+        external
+        payable
+        nonReentrant
+    {
+        StrategyStruct.Proposal storage proposal = proposals[proposalId];
+
+        if (address(arbitrableConfig.arbitrator) == address(0)) {
+            revert ArbitratorCannotBeZero();
+        }
+        if (address(collateralVault) == address(0)) {
+            revert CollateralVaultCannotBeZero();
+        }
+        if (proposal.proposalId != proposalId) {
+            revert ProposalNotInList(proposalId);
+        }
+        if (proposal.proposalStatus != StrategyStruct.ProposalStatus.Active) {
+            revert ProposalNotActive(proposalId);
+        }
+        if (msg.value <= arbitrableConfig.collateralAmount) {
+            revert InsufficientCollateral(msg.value, arbitrableConfig.collateralAmount);
+        }
+
+        uint256 arbitrationFee = msg.value - arbitrableConfig.collateralAmount;
+
+        collateralVault.depositCollateral{value: arbitrableConfig.collateralAmount}(proposalId, msg.sender);
+
+        uint256 disputeId = arbitrableConfig.arbitrator.createDispute{value: arbitrationFee}(RULING_OPTIONS, _extraData);
+
+        proposal.proposalStatus = StrategyStruct.ProposalStatus.Disputed;
+        proposal.disputeId = disputeId;
+        proposal.disputeTimestamp = block.timestamp;
+        proposal.challenger = msg.sender;
+        disputeIdToProposalId[disputeId] = proposalId;
+        emit ProposalDisputed(proposalId, disputeId, msg.sender, arbitrationFee, context);
+    }
+
+    function rule(uint256 _disputeID, uint256 _ruling) external override {
+        uint256 proposalId = disputeIdToProposalId[_disputeID];
+        StrategyStruct.Proposal storage proposal = proposals[proposalId];
+
+        if (proposalId == 0) {
+            revert ProposalNotInList(proposalId);
+        }
+        if (proposal.proposalStatus != StrategyStruct.ProposalStatus.Disputed) {
+            revert ProposalNotDisputed(proposalId);
+        }
+
+        bool isTimeOut = block.timestamp > proposal.disputeTimestamp + arbitrableConfig.defaultRulingTimeout;
+
+        if (!isTimeOut && msg.sender != address(arbitrableConfig.arbitrator)) {
+            revert OnlyArbitrator();
+        }
+
+        if (isTimeOut || _ruling == 0) {
```

Comment:
Should we revert if not arbitrator and not abstain ruling  ?

Reply:
in which case you think we should check if not arbitrator? i mean the rule function will be called by the arbitrator or by anyone, if is called by anyone doesnt matter if there is arbitrator or not

Reply:
I forgot to add revert when timeout, not arbitrator, and not abstain. Because if you can pass the ruling, a potential lambda caller will call this function and apply the outcome when abstain whenever they passed approve or reject. My point is maybe he didnt wanted to apply the abstain outcome and preffer waiting before ruling the dispute, for me it would be a totally unexpected side effect that can lead to regrets of the caller.

Reply:
So basically what im suggesting is we should not force a ruling after timeout we should revert if its not the expected ruling instead.
And now that I realized Arbitrator are also expected to rule with abstain after tmeout then this revert condition would be if(Timeout && ruling not abstain)